src/deluge/gui/views/automation_clip_view.cpp

```diff
@@ -300,144 +298,114 @@ bool AutomationInstrumentClipView::opened() {
 	initPadSelection();
 
 	InstrumentClip* clip = getCurrentInstrumentClip();
-	Instrument* instrument = (Instrument*)clip->output;
 
-	//check if we for some reason, left the automation view, then switched clip types, then came back in
-	//if you did that...reset the parameter selection and save the current parameter type selection
-	//so we can check this again next time it happens
-	if (instrument->type != clip->lastSelectedOutputType) {
-		initParameterSelection();
+	//only applies to instrument clips (not audio)
+	if (clip) {
+		OutputType outputType = clip->output->type;
+		//check if we for some reason, left the automation view, then switched clip types, then came back in
+		//if you did that...reset the parameter selection and save the current parameter type selection
+		//so we can check this again next time it happens
+		if (outputType != clip->lastSelectedOutputType) {
+			initParameterSelection();
```

Comment:
> Is there any similar sort of reset required for audio clips?

No because when you're in an audio clip, you can't switch clip types.

In an instrument clip you can be in a synth and switch the type to a kit, midi, cv while you are in the clip which necessitates a reset.

The same behaviour does not exist in an audio clip